### DISCUSSION TITLE
How to serialize augmentation parameters?

### DISCUSSION THREAD

**Discussion by oeway** [kornia/kornia#1745] [Questions] [answered]
In albumentations, it support serialize the parameters into a JSON or YAML file (https://albumentations.ai/docs/examples/serialization/) which is neat to keep track of the augmentation and also great for sending config files to servers to create augmentation pipeline. I am wondering if we can do the same with kornia.

**Comment by edgarriba** [comment, marked as answer]
@oeway every augmentation is a `nn.Module` -- so you can save as a regular pytorch module as follows 

```python
import torch
import kornia as K

aug1 = K.augmentation.RandomRotation(90)
out = aug1(torch.rand(2, 3, 4, 5))

torch.save(aug1, "aug.pt")
aug2 = torch.load("aug.pt")
assert_close(aug1._params, aug2._params)
```

Colab here: https://colab.research.google.com/drive/1YxNM8ERabTjzq2V5yJHuEXvB7NYFpZym?usp=sharing

**Reply by edgarriba** [reply]
@shijianjian btw i found that the params are not always batched. Not sure if i'm missing something

**Reply by oeway** [reply]
Oh I see, so I can also save it as  torchscript.

In terms of security, maybe it's not safe to execute user submitted torch modules or torchscript? Ideally we would like to have a json like approach which should be safer.

**Reply by shijianjian** [reply]
torchscript is not well-supported at the moment. For parsing from dict or json, you may refer to https://github.com/ternaus/iglovikov_helper_functions/blob/master/iglovikov_helper_functions/config_parsing/utils.py.

It should be simply doerable. A PR is welcomed.

**Reply by edgarriba** [reply]
@shijianjian I’d be more in favour to support that from our augmentations modules we can do the following 
```python 
torch.save(aug.state_dict(), PATH)

new_aug.load_state_dict(torch.load(PATH))


```
